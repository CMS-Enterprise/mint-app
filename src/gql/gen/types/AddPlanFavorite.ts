/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: AddPlanFavorite
// ====================================================

export interface AddPlanFavorite_addPlanFavorite {
  __typename: "PlanFavorite";
  modelPlanID: UUID;
  userID: UUID;
}

export interface AddPlanFavorite {
  addPlanFavorite: AddPlanFavorite_addPlanFavorite;
}

export interface AddPlanFavoriteVariables {
  modelPlanID: UUID;
}
