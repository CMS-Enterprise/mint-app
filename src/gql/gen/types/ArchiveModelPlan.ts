/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: ArchiveModelPlan
// ====================================================

export interface ArchiveModelPlan_updateModelPlan {
  __typename: "ModelPlan";
  id: UUID;
}

export interface ArchiveModelPlan {
  updateModelPlan: ArchiveModelPlan_updateModelPlan;
}

export interface ArchiveModelPlanVariables {
  id: UUID;
  archived: boolean;
}
