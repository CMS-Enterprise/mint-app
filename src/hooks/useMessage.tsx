import React, {
  createContext,
  ReactNode,
  useContext,
  useEffect,
  useState
} from 'react';
import { useLocation } from 'react-router-dom';

const MessageContext = createContext<
  | {
      message: string | React.ReactNode | undefined;
      showMessage: (message: string | React.ReactNode) => void;
      showMessageOnNextPage: (message: string | React.ReactNode) => void;
    }
  | undefined
>(undefined);

// MessageProvider manages the state necessary for child components to
// use the useMessage hook.
const MessageProvider = ({ children }: { children: ReactNode }) => {
  const [queuedMessage, setQueuedMessage] = useState<
    string | React.ReactNode
  >();
  const [message, setMessage] = useState<string | React.ReactNode>();
  const location = useLocation();

  const [lastPathname, setLastPathname] = useState(location.pathname);

  useEffect(() => {
    if (lastPathname !== location.pathname) {
      setMessage(queuedMessage);
      setQueuedMessage('');
      setLastPathname(location.pathname);
    }
  }, [message, queuedMessage, lastPathname, location.pathname]);

  return (
    <MessageContext.Provider
      value={{
        message,
        showMessage: setMessage,
        showMessageOnNextPage: setQueuedMessage
      }}
    >
      {children}
    </MessageContext.Provider>
  );
};

// useMessage is a hook that provides the ability to show notifications to the
// user. Components will use the hook to get access to two methods:
//
// showMessage: display a message to the user immediately. This is useful
// when an interaction won't cause an immediate redirect to a new page.
//
// showMessageOnNextPage: schedule a message to be shown on the next
// visited by the user. Pages are identified using their URL and therefore
// messages won't appear until the URL changes.
//
// Currently there is only support for showing a single message.
const useMessage = () => {
  const context = useContext(MessageContext);
  if (context === undefined) {
    throw new Error('useMessage must be used within a MessageProvider');
  }

  return context;
};

export default useMessage;
export { useMessage, MessageProvider };
