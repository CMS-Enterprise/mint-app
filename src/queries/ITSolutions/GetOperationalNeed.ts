import { gql } from '@apollo/client';

export default gql`
  query GetOperationalNeed($id: UUID!, $includeNotNeeded: Boolean = true) {
    operationalNeed(id: $id) {
      id
      modelPlanID
      name
      key
      nameOther
      needed
      solutions(includeNotNeeded: $includeNotNeeded) {
        id
        isOther
        name
        key
        pocName
        pocEmail
        needed
        nameOther
        mustStartDts
        mustFinishDts
        status
      }
    }
  }
`;
