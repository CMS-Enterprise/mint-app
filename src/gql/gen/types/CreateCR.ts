/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { PlanCRCreateInput } from "./../../../types/graphql-global-types";

// ====================================================
// GraphQL mutation operation: CreateCR
// ====================================================

export interface CreateCR_createPlanCR {
  __typename: "PlanCR";
  id: UUID;
  modelPlanID: UUID;
  idNumber: string;
  dateInitiated: Time;
  dateImplemented: Time | null;
  title: string;
  note: string | null;
}

export interface CreateCR {
  createPlanCR: CreateCR_createPlanCR;
}

export interface CreateCRVariables {
  input: PlanCRCreateInput;
}
