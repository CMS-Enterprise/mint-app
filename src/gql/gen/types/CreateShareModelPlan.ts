/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ModelViewFilter } from "./../../../types/graphql-global-types";

// ====================================================
// GraphQL mutation operation: CreateShareModelPlan
// ====================================================

export interface CreateShareModelPlan {
  shareModelPlan: boolean;
}

export interface CreateShareModelPlanVariables {
  modelPlanID: UUID;
  viewFilter?: ModelViewFilter | null;
  usernames: string[];
  optionalMessage?: string | null;
}
