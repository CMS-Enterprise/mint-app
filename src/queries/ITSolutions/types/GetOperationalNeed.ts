/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { OperationalNeedKey, OperationalSolutionKey, OpSolutionStatus } from "./../../../types/graphql-global-types";

// ====================================================
// GraphQL query operation: GetOperationalNeed
// ====================================================

export interface GetOperationalNeed_operationalNeed_solutions {
  __typename: "OperationalSolution";
  id: UUID;
  name: string | null;
  key: OperationalSolutionKey | null;
  pocName: string | null;
  pocEmail: string | null;
  needed: boolean | null;
  nameOther: string | null;
  isOther: boolean;
  otherHeader: string | null;
  mustStartDts: Time | null;
  mustFinishDts: Time | null;
  status: OpSolutionStatus;
}

export interface GetOperationalNeed_operationalNeed {
  __typename: "OperationalNeed";
  id: UUID;
  modelPlanID: UUID;
  name: string | null;
  key: OperationalNeedKey | null;
  nameOther: string | null;
  needed: boolean | null;
  solutions: GetOperationalNeed_operationalNeed_solutions[];
}

export interface GetOperationalNeed {
  operationalNeed: GetOperationalNeed_operationalNeed;
}

export interface GetOperationalNeedVariables {
  id: UUID;
  includeNotNeeded?: boolean | null;
}
