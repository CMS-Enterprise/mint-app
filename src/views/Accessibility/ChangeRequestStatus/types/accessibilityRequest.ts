/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AccessibilityRequestStatus } from "./../../../../types/graphql-global-types";

// ====================================================
// GraphQL query operation: accessibilityRequest
// ====================================================

export interface accessibilityRequest_accessibilityRequest_statusRecord {
  __typename: "AccessibilityRequestStatusRecord";
  status: AccessibilityRequestStatus;
}

export interface accessibilityRequest_accessibilityRequest {
  __typename: "AccessibilityRequest";
  id: UUID;
  name: string;
  statusRecord: accessibilityRequest_accessibilityRequest_statusRecord;
}

export interface accessibilityRequest {
  accessibilityRequest: accessibilityRequest_accessibilityRequest | null;
}

export interface accessibilityRequestVariables {
  id: UUID;
}
