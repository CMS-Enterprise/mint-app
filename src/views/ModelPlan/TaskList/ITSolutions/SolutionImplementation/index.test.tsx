import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { MessageProvider } from 'hooks/useMessage';
import GetOperationalNeed from 'queries/ITSolutions/GetOperationalNeed';
import { GetOperationalNeed_operationalNeed as GetOperationalNeedType } from 'queries/ITSolutions/types/GetOperationalNeed';
import {
  OperationalNeedKey,
  OperationalSolutionKey,
  OpSolutionStatus
} from 'types/graphql-global-types';

import SolutionImplmentation from '.';

const modelID = 'ce3405a0-3399-4e3a-88d7-3cfc613d2905';
const operationalNeedID = '081cb879-bd6f-4ead-b9cb-3a299de76390';

const operationalNeed: GetOperationalNeedType = {
  __typename: 'OperationalNeed',
  id: operationalNeedID,
  modelPlanID: modelID,
  name: 'Obtain an application support contractor',
  key: OperationalNeedKey.APP_SUPPORT_CON,
  nameOther: null,
  needed: true,
  solutions: [
    {
      __typename: 'OperationalSolution',
      id: '00000000-0000-0000-0000-000000000000',
      name: 'Research, Measurement, Assessment, Design, and Analysis (RMADA)',
      pocEmail: '',
      key: OperationalSolutionKey.RMADA,
      mustStartDts: null,
      mustFinishDts: null,
      isOther: false,
      isCommonSolution: true,
      otherHeader: null,
      status: OpSolutionStatus.AT_RISK,
      needed: true,
      pocName: 'John Doe',
      nameOther: null
    }
  ]
};

const mocks = [
  {
    request: {
      query: GetOperationalNeed,
      variables: {
        id: operationalNeedID,
        includeNotNeeded: false
      }
    },
    result: {
      data: {
        operationalNeed
      }
    }
  }
];

describe('IT Solutions NeedQuestionAndAnswer', () => {
  it('renders correctly', async () => {
    const { getByText, getAllByTestId, getByRole } = render(
      <MemoryRouter
        initialEntries={[
          {
            pathname: `/models/${modelID}/task-list/it-solutions/${operationalNeedID}/solution-implementation-details`
          }
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <Route path="/models/:modelID/task-list/it-solutions/:operationalNeedID/solution-implementation-details">
            <MessageProvider>
              <SolutionImplmentation />
            </MessageProvider>
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(
        getByText('Research, Measurement, Assessment, Design, and Analysis')
      ).toBeInTheDocument();
    });

    const datePicker = getAllByTestId('date-picker-external-input')[0];
    userEvent.type(datePicker, '12/10/2030');

    await waitFor(() => {
      expect(datePicker).toHaveValue('12/10/2030');
    });

    const atRisk = getByRole('radio', { name: 'At risk' });
    const backlog = getByRole('radio', { name: 'Backlog' });
    userEvent.click(backlog);

    await waitFor(() => {
      expect(atRisk).not.toBeChecked();
      expect(backlog).toBeChecked();
    });
  });

  it('matches snapshot', async () => {
    const { asFragment, getByText } = render(
      <MemoryRouter
        initialEntries={[
          {
            pathname: `/models/${modelID}/task-list/it-solutions/${operationalNeedID}/solution-implementation-details`
          }
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <Route path="/models/:modelID/task-list/it-solutions/:operationalNeedID/solution-implementation-details">
            <MessageProvider>
              <SolutionImplmentation />
            </MessageProvider>
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(
        getByText('Research, Measurement, Assessment, Design, and Analysis')
      ).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
