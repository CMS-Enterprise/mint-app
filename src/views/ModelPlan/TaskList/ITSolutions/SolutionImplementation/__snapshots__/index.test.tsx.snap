// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Operational Solutions NeedQuestionAndAnswer > matches snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumbs"
    class="usa-breadcrumb usa-breadcrumb--wrap"
  >
    <ol
      class="usa-breadcrumb__list"
    >
      <li
        class="usa-breadcrumb__list-item"
      >
        <a
          class="usa-link usa-breadcrumb__link"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
      </li>
      <li
        class="usa-breadcrumb__list-item"
      >
        <a
          class="usa-link usa-breadcrumb__link"
          href="/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/collaboration-area/task-list/"
        >
          <span>
            Model Plan task list
          </span>
        </a>
      </li>
      <li
        class="usa-breadcrumb__list-item"
      >
        <a
          class="usa-link usa-breadcrumb__link"
          href="/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/collaboration-area/task-list/it-solutions"
        >
          <span>
            Operational solutions tracker
          </span>
        </a>
      </li>
      <li
        aria-current="page"
        class="usa-breadcrumb__list-item usa-current"
      >
        <span>
          Select this solution
        </span>
      </li>
    </ol>
  </nav>
  <div
    class="grid-row grid-gap"
    data-testid="grid"
  >
    <div
      class="tablet:grid-col-9"
      data-testid="grid"
    >
      <h1
        aria-live="polite"
        class="mint-h1 margin-top-4 margin-bottom-2"
        tabindex="-1"
      >
        Select this solution
      </h1>
      <p
        class="margin-top-0 margin-bottom-1 font-body-lg"
        data-testid="model-plan-name"
      >
        for 
      </p>
      <p
        class="line-height-body-4"
      >
        Add additional information and update the implementation status of each selected solution. If you do not know this information you may add it at a later date.
      </p>
      <div
        class="tablet:grid-col-12 desktop:grid-col-8"
        data-testid="grid"
      >
        <div
          class="grid-container padding-3 bg-base-lightest maxw-none"
          data-testid="gridContainer"
        >
          <div
            class="grid-row"
            data-testid="grid"
          >
            <div
              class="desktop:grid-col-12"
              data-testid="grid"
            >
              <p
                class="text-bold margin-y-1"
              >
                Operational need
              </p>
              <p
                class="margin-top-0 font-body-sm"
              >
                Obtain an application support contractor
              </p>
            </div>
            <div
              class="desktop:grid-col-12"
              data-testid="grid"
            >
              <button
                class="usa-button usa-button--unstyled display-flex flex-align-center text-ls-1 deep-underline margin-bottom-1 margin-top-1"
                data-testid="toggle-need-answer"
                type="button"
              >
                <svg
                  class="usa-icon margin-right-05 needs-question__rotate"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m12 8-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
                  />
                </svg>
                Why is this a part of my operational needs?
              </button>
            </div>
          </div>
        </div>
        <form
          action="#"
          class="margin-top-6"
        >
          <fieldset
            class="usa-fieldset"
            data-testid="fieldset"
          >
            <div>
              <p
                class="text-bold"
              >
                Solution
              </p>
              <ul
                class="usa-card-group flex-column flex-no-wrap"
                data-testid="CardGroup"
              >
                <li
                  class="usa-card solution-card shadow"
                  data-testid="Card"
                >
                  <div
                    class="usa-card__container"
                  >
                    <div
                      class="padding-3"
                    >
                      <h3
                        class="margin-top-0 solutions-checkbox__header margin-bottom-1"
                      >
                        Research, Measurement, Assessment, Design, and Analysis
                      </h3>
                      <h5
                        class="margin-top-0 margin-bottom-2 text-normal"
                      >
                        (RMADA)
                      </h5>
                      <div
                        class="tablet:grid-col-6 margin-bottom-2"
                        data-testid="grid"
                      >
                        <p
                          class="text-bold margin-bottom-0"
                        >
                          Point of contact
                        </p>
                        <p
                          class="margin-y-0"
                        >
                          Alicia Thomas
                        </p>
                        <a
                          aria-label="Send an email"
                          class="usa-link line-height-body-5 display-flex flex-align-center"
                          href="mailto:at.mint@oddball.io"
                          target="_blank"
                        >
                          <div>
                            at.mint@oddball.io
                          </div>
                          <svg
                            class="usa-icon margin-left-05 text-tbottom"
                            focusable="false"
                            height="1em"
                            role="img"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                            />
                          </svg>
                        </a>
                      </div>
                      <div
                        class="mint-divider"
                      />
                      <a
                        class="usa-link display-flex flex-align-center usa-button usa-button--unstyled margin-top-2"
                        href="/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/collaboration-area/task-list/it-solutions/081cb879-bd6f-4ead-b9cb-3a299de76390/solution-implementation-details?solution=research-measurement-assessment-design-and-analysis&section=about"
                      >
                        About this solution
                        <svg
                          class="usa-icon margin-left-1"
                          focusable="false"
                          height="1em"
                          role="img"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                          />
                        </svg>
                      </a>
                    </div>
                  </div>
                </li>
              </ul>
              <div
                class="usa-form-group margin-top-1"
                data-scroll="mustStartDts"
              >
                <label
                  class="usa-label text-bold"
                  data-testid="label"
                  for="solution-must-start-RMADA"
                >
                  Must start by
                </label>
                <div
                  class="usa-hint margin-top-1"
                >
                  mm/dd/yyyy
                </div>
                <div
                  class="width-card-lg position-relative"
                >
                  <div
                    class="usa-date-picker usa-date-picker--initialized"
                    data-testid="date-picker"
                  >
                    <input
                      aria-hidden="true"
                      class="usa-input usa-sr-only usa-date-picker__internal-input"
                      data-testid="date-picker-internal-input"
                      error="0"
                      maxlength="50"
                      name="solutions[0].mustStartDts"
                      readonly=""
                      tabindex="-1"
                      type="text"
                      value=""
                    />
                    <div
                      class="usa-date-picker__wrapper"
                      tabindex="-1"
                    >
                      <input
                        class="usa-input usa-date-picker__external-input"
                        data-testid="date-picker-external-input"
                        error="0"
                        id="solution-must-start-RMADA"
                        maxlength="50"
                        type="text"
                        value=""
                      />
                      <button
                        aria-haspopup="true"
                        aria-label="Toggle calendar"
                        class="usa-date-picker__button"
                        data-testid="date-picker-button"
                        type="button"
                      />
                      <div
                        aria-modal="true"
                        class="usa-date-picker__calendar"
                        data-testid="date-picker-calendar"
                        hidden=""
                        role="dialog"
                        style="top: 0px;"
                      />
                      <div
                        aria-live="polite"
                        class="usa-sr-only usa-date-picker__status"
                        data-testid="date-picker-status"
                        role="status"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="usa-form-group"
                data-scroll="mustFinishDts"
              >
                <label
                  class="usa-label text-bold"
                  data-testid="label"
                  for="solution-must-finish-RMADA"
                >
                  Must be ready by
                </label>
                <div
                  class="usa-hint margin-top-1"
                >
                  mm/dd/yyyy
                </div>
                <div
                  class="width-card-lg position-relative"
                >
                  <div
                    class="usa-date-picker usa-date-picker--initialized"
                    data-testid="date-picker"
                  >
                    <input
                      aria-hidden="true"
                      class="usa-input usa-sr-only usa-date-picker__internal-input"
                      data-testid="date-picker-internal-input"
                      error="0"
                      maxlength="50"
                      name="solutions[0].mustFinishDts"
                      readonly=""
                      tabindex="-1"
                      type="text"
                      value=""
                    />
                    <div
                      class="usa-date-picker__wrapper"
                      tabindex="-1"
                    >
                      <input
                        class="usa-input usa-date-picker__external-input"
                        data-testid="date-picker-external-input"
                        error="0"
                        id="solution-must-finish-RMADA"
                        maxlength="50"
                        type="text"
                        value=""
                      />
                      <button
                        aria-haspopup="true"
                        aria-label="Toggle calendar"
                        class="usa-date-picker__button"
                        data-testid="date-picker-button"
                        type="button"
                      />
                      <div
                        aria-modal="true"
                        class="usa-date-picker__calendar"
                        data-testid="date-picker-calendar"
                        hidden=""
                        role="dialog"
                        style="top: 0px;"
                      />
                      <div
                        aria-live="polite"
                        class="usa-sr-only usa-date-picker__status"
                        data-testid="date-picker-status"
                        role="status"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="usa-form-group"
              >
                <label
                  class="usa-label text-bold"
                  data-testid="label"
                  for="solution-status-RMADA"
                >
                  What is the status of this solution?
                </label>
                <fieldset
                  class="usa-fieldset"
                  data-testid="fieldset"
                >
                  <div
                    class="usa-radio"
                    data-testid="radio"
                  >
                    <input
                      class="usa-radio__input"
                      id="solution-status-RMADA-NOT_STARTED"
                      name="solutions[0].status"
                      type="radio"
                      value="NOT_STARTED"
                    />
                    <label
                      class="usa-radio__label"
                      for="solution-status-RMADA-NOT_STARTED"
                    >
                      Not started
                    </label>
                  </div>
                  <div
                    class="usa-radio"
                    data-testid="radio"
                  >
                    <input
                      class="usa-radio__input"
                      id="solution-status-RMADA-ONBOARDING"
                      name="solutions[0].status"
                      type="radio"
                      value="ONBOARDING"
                    />
                    <label
                      class="usa-radio__label"
                      for="solution-status-RMADA-ONBOARDING"
                    >
                      Onboarding
                    </label>
                  </div>
                  <div
                    class="usa-radio"
                    data-testid="radio"
                  >
                    <input
                      class="usa-radio__input"
                      id="solution-status-RMADA-BACKLOG"
                      name="solutions[0].status"
                      type="radio"
                      value="BACKLOG"
                    />
                    <label
                      class="usa-radio__label"
                      for="solution-status-RMADA-BACKLOG"
                    >
                      Backlog
                    </label>
                  </div>
                  <div
                    class="usa-radio"
                    data-testid="radio"
                  >
                    <input
                      class="usa-radio__input"
                      id="solution-status-RMADA-IN_PROGRESS"
                      name="solutions[0].status"
                      type="radio"
                      value="IN_PROGRESS"
                    />
                    <label
                      class="usa-radio__label"
                      for="solution-status-RMADA-IN_PROGRESS"
                    >
                      In progress
                    </label>
                  </div>
                  <div
                    class="usa-radio"
                    data-testid="radio"
                  >
                    <input
                      class="usa-radio__input"
                      id="solution-status-RMADA-COMPLETED"
                      name="solutions[0].status"
                      type="radio"
                      value="COMPLETED"
                    />
                    <label
                      class="usa-radio__label"
                      for="solution-status-RMADA-COMPLETED"
                    >
                      Completed
                    </label>
                  </div>
                  <div
                    class="usa-radio"
                    data-testid="radio"
                  >
                    <input
                      class="usa-radio__input"
                      id="solution-status-RMADA-AT_RISK"
                      name="solutions[0].status"
                      type="radio"
                      value="AT_RISK"
                    />
                    <label
                      class="usa-radio__label"
                      for="solution-status-RMADA-AT_RISK"
                    >
                      At risk
                    </label>
                  </div>
                </fieldset>
              </div>
              <div
                class="desktop:grid-col-12"
                data-testid="grid"
              >
                <button
                  class="usa-button usa-button--unstyled display-flex flex-align-center text-ls-1 deep-underline margin-bottom-1 margin-top-3"
                  data-testid="toggle-need-answer"
                  type="button"
                >
                  <svg
                    class="usa-icon margin-right-05 needs-question__rotate"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m12 8-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
                    />
                  </svg>
                  Implementation statuses
                </button>
              </div>
            </div>
            <div
              class="margin-top-6 margin-bottom-3"
            >
              <button
                class="usa-button usa-button usa-button--outline margin-bottom-1"
                data-testid="button"
                type="button"
              >
                Back
              </button>
              <button
                class="usa-button"
                data-testid="button"
                id="submit-solutions"
                type="submit"
              >
                Save solutions
              </button>
            </div>
            <button
              class="usa-button usa-button usa-button--unstyled display-flex flex-align-center margin-bottom-6"
              data-testid="button"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="usa-icon margin-right-1"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
              Don’t add solutions and return to tracker
            </button>
          </fieldset>
        </form>
      </div>
    </div>
    <div
      class="desktop:grid-col-3 padding-x-1"
      data-testid="grid"
    >
      <div
        class="border-top-05 border-primary-lighter padding-top-2 margin-top-4"
      >
        <div
          class="padding-2 bg-primary-lighter"
        >
          <p
            class="text-bold margin-top-0"
          >
            Need help choosing a solution?
          </p>
          <div
            class="display-flex"
            data-testid="ask-a-question"
          >
            <svg
              class="usa-icon text-primary margin-right-1"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
              />
            </svg>
            <button
              class="usa-button usa-button--unstyled"
              data-testid="ask-a-question-button"
              type="button"
            >
              Start a discussion
            </button>
          </div>
        </div>
      </div>
      <div
        class="margin-top-4"
      >
        <p
          class="text-bold margin-bottom-2"
        >
          Helpful links
        </p>
        <button
          class="usa-button usa-button usa-button--unstyled line-height-body-5"
          data-testid="button"
          type="button"
        >
          Available operational solutions (opens in a new tab)
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
